ImageFeed/ImagesList/ImagesListService.swift

```diff
@@ -0,0 +1,173 @@
+    static let DidChangeNotification = Notification.Name(rawValue: "ImagesListServiceDidChange")
+
+    private lazy var dateFormatter: DateFormatter = {
+        let formatter = DateFormatter()
+        formatter.dateStyle = .long
+        formatter.timeStyle = .none
+        return formatter
+    }()
+
+    private func makePhotosRequest(pageNumber: Int, token: String) -> URLRequest {
+        var urlComponents = URLComponents(string: Constants.getPhotosUrlString)!
+        urlComponents.queryItems = [
+            URLQueryItem(name: "page", value: String(pageNumber)),
+            URLQueryItem(name: "per_page", value: "10"),
+        ]
+
+        guard let url = urlComponents.url else {
+            fatalError("makeRequest Error")
+        }
+        var request = URLRequest(url: url)
+        request.httpMethod = "GET"
+        request.setValue("Bearer \(token)", forHTTPHeaderField: "Authorization")
+        return request
+    }
+
+    private func makeLikeChangingRequest(photoId: String, isLike: Bool, token: String) -> URLRequest {
+        let urlComponents = URLComponents(string: Constants.getPhotosUrlString)!
+        guard var url = urlComponents.url else {
+            fatalError("makeRequest Error")
+        }
+        url.appendPathComponent("/\(photoId)/like")
+        var request = URLRequest(url: url)
+        request.httpMethod = isLike ? "POST" : "DELETE"
+        request.setValue("Bearer \(token)", forHTTPHeaderField: "Authorization")
+        return request
+    }
+    
+    func fetchPhotosNextPage(token: String) {
+        let fulfillCompletionOnMainThread: (Result<[PhotoResult], Error>) -> Void = { result in
+            DispatchQueue.main.async {
+                switch result {
+                case .success(let images):
+                    for image in images {
+                        let convertedImage = Photo(
+                            id: image.id,
+                            size: CGSize(width: image.width, height: image.height),
+                            createdAt: self.dateFormatter.date(from: image.createdAt),
+                            welcomeDescription:image.welcomeDescription,
+                            thumbImageURL: image.urls.thumb,
+                            largeImageURL: image.urls.full,
+                            isLiked: image.isLiked
+                        )
+                        self.photos.append(convertedImage);
+                    }
+                    self.task = nil
+                    self.lastLoadedPage += 1
+                    self.notificationCenter
+                        .post(
+                            name: ImageListService.DidChangeNotification,
+                            object: self,
+                            userInfo: ["PHOTOS": self.photos])
+                case .failure(_):
+                    self.task = nil
+                    return
+                }
+            }
+        }
+
+        if self.task != nil {return}
```

Comment:
Фигурыные скобки сопровождаются пробелами
```suggestion
        if self.task != nil { return }
```